simple_salesforce/api.py

```diff
@@ -603,8 +603,10 @@ def _call_salesforce(self, method, url, name="", **kwargs):
 
         if self._salesforce_login_partial is not None \
                 and result.status_code == 401:
-            self._refresh_session()
-            return self._call_salesforce(method, url, name, **kwargs)
+            error_details = json.loads(result.text.lstrip('[').rstrip(']'))
+            if error_details['errorCode'] == 'INVALID_SESSION_ID':
```

Comment:
I know Salesforce can have two different types of response here, depending on which endpoint was used: returning a list or just one object directly. But are you sure the list will always only have one object? Otherwise removing the `[]` JSON array characters can yield an unparseable JSON.

Reply:
According the [developer guide](https://developer.salesforce.com/docs/atlas.en-us.240.0.api_rest.meta/api_rest/errorcodes.htm), the response seems to have the array chars.

Reply:
Is there a reason not to decode the JSON list and get the first item, in case there are multiple items?

For example:
```python
error_list = json.loads(result.text)
error_details = error_list[0]  # this could be combined with either of the other statements
if error_details['errorCode'] == 'INVALID_SESSION_ID':
```

Reply:
This can be simplified to:
```python
            response_content = result.json()
            if response_content[0]['errorCode'] == 'INVALID_SESSION_ID':
```